lib/httplib/httplib.go

```diff
@@ -69,6 +72,20 @@ func MakeStdHandler(fn StdHandlerFunc) http.HandlerFunc {
 	}
 }
 
+// WithCSRFProtection ensures that request to unauthenticated API is checked against CSRF attacks
+func WithCSRFProtection(fn HandlerFunc) httprouter.Handle {
+	hanlderFn := MakeHandler(fn)
+	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		err := csrf.VerifyHTTPHeader(r)
```

Comment:
so... we have `err` here which is supposed to be useful and informative (it is returned from a higher level funcion `VerifyHTTPHeader()`. but `err` is not used, and instead is wrapped (below) into another two different (again, why different?) error messages. why not have  `VerifyHTTPHeader()` return something which is good enough to pass directly to the log and return on the wire?